In [1]:
#IMPORT LIBRARIES

#Scipy Libraries
import numpy as np
import pandas as pd
import scipy.stats as stats
import statsmodels.api as sm
import matplotlib.pyplot as plt

#Seaborn library
import seaborn as sns


Dataset downloaded from here: https://datadashboard.fda.gov/ora/cd/inspections.htm

Specifically the inspection citations dataset

In [128]:
insp = pd.read_csv('FDA_Inspections.csv', parse_dates=['Inspection End Date'], index_col= ['Inspection End Date'])#Read in the Data
insp = pd.DataFrame(insp)#Create a dataframe

#drop the Inspection ID and FEI number columns as they are mixed data types and not relevant to the analysis
insp=insp.drop(['Inspection ID', 'FEI Number'], axis=1)

print(insp.shape)
print(insp.columns)

(220595, 6)
Index(['Legal Name', 'Program Area', 'Act/CFR Number', 'Short Description',
       'Long Description', 'Firm Profile'],
      dtype='object')


In [129]:
insp.head(5)

Legal Name Program Area  \
Inspection End Date                                              
2021-06-16                        ARMS GLOBAL INC        Foods   
2021-06-15           Mesada International Trading        Foods   
2021-06-10                            ImaCor Inc.      Devices   
2021-06-10                            ImaCor Inc.      Devices   
2021-06-10                            ImaCor Inc.      Devices   

                        Act/CFR Number  \
Inspection End Date                      
2021-06-16             21 CFR 1.502(a)   
2021-06-15             21 CFR 1.502(a)   
2021-06-10               21 CFR 803.17   
2021-06-10            21 CFR 820.40(a)   
2021-06-10           21 CFR 820.198(a)   

                                           Short Description  \
Inspection End Date                                            
2021-06-16                                      Develop FSVP   
2021-06-15                                      Develop FSVP   
2021-06-10                    Lack of Written MDR Procedures   
2021-06-10           Document review, approval documentation   
2021-06-10                                        Complaints   

                                                      Long Description  \
Inspection End Date                                                      
2021-06-16                                You did not develop an FSVP.   
2021-06-15                                You did not develop an FSVP.   
2021-06-10            Written MDR procedures have not been maintained.   
2021-06-10           The documentation of approval of documents doe...   
2021-06-10              Complaint files are not adequately maintained.   

                                                          Firm Profile  
Inspection End Date                                                     
2021-06-16           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-15           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-10           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-10           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-10           https://datadashboard.fda.gov/ora/firmprofile....

### SPLIT THE DATASET INTO EACH INSPECTION CATEGORY

LIST EACH INSPECTION CATEGORY

In [130]:
#Dropping duplicates will show each unique area of inspection
unique_cats = insp.drop_duplicates(subset = ["Program Area"])
unique_cats

Legal Name  \
Inspection End Date                                                      
2021-06-16                                             ARMS GLOBAL INC   
2021-06-10                                                 ImaCor Inc.   
2021-05-28                                         Bravo Packing, Inc.   
2021-05-21                       Fresenius Medical Care, North America   
2021-05-18           U.S. Dept. of Veterans Affairs, Northport VA M...   
2021-05-03                                          Chafic Karam, M.D.   
2021-03-16                                  Southwest Fertility Center   
2021-02-25           Modesto Food Distributors dba Onorato & Compan...   
2020-03-05                                Mevion Medical Systems, Inc.   
2020-02-11                                   Borenstein Caterers, Inc.   

                                         Program Area     Act/CFR Number  \
Inspection End Date                                                        
2021-06-16                                      Foods    21 CFR 1.502(a)   
2021-06-10                                    Devices      21 CFR 803.17   
2021-05-28                        Veterinary Medicine      21 CFR 507.14   
2021-05-21                                      Drugs     21 CFR 211.192   
2021-05-18                                  Biologics   21 CFR 600.12(a)   
2021-05-03                     Bioresearch Monitoring      21 CFR 312.60   
2021-03-16           Human Tissue for Transplantation  21 CFR 1271.85(a)   
2021-02-25                         Part 11 Compliance                NaN   
2020-03-05                          Radiologic Health     21 CFR 1002.13   
2020-02-11                        Parts 1240 and 1250  21 CFR 1250.30(d)   

                                              Short Description  \
Inspection End Date                                               
2021-06-16                                         Develop FSVP   
2021-06-10                       Lack of Written MDR Procedures   
2021-05-28                                            Personnel   
2021-05-21           Written record of investigation incomplete   
2021-05-18                           Maintenance  - concurrence   
2021-05-03                         FD-1572, protocol compliance   
2021-03-16           Infection with communicable disease agents   
2021-02-25                              Procedural requirements   
2020-03-05                                    Failure to submit   
2020-02-11           Plumbing design, installation, maintenance   

                                                      Long Description  \
Inspection End Date                                                      
2021-06-16                                You did not develop an FSVP.   
2021-06-10            Written MDR procedures have not been maintained.   
2021-05-28           You did not take reasonable measures and preca...   
2021-05-21           Written records of investigations into unexpla...   
2021-05-18           Records are not made concurrently with the per...   
2021-05-03           An investigation was not conducted in accordan...   
2021-03-16           Donors were not tested for evidence of infecti...   
2021-02-25           Deviation from the procedural requirements of ...   
2020-03-05           You did not submit an annual report for produc...   
2020-02-11           Plumbing is not maintained so as to prevent co...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-06-16           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-10           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-28           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-21           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-18           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-03           https://datadashboard.fda.gov/or

In [131]:
Foods = insp[insp["Program Area"] == 'Foods']
Foods.head(5)

Legal Name Program Area  \
Inspection End Date                                                 
2021-06-16                           ARMS GLOBAL INC        Foods   
2021-06-15              Mesada International Trading        Foods   
2021-06-09           Fresh Goods Trading Corporation        Foods   
2021-06-09                            Mutti USA Inc.        Foods   
2021-06-07                          Oax Imports Corp        Foods   

                      Act/CFR Number Short Description  \
Inspection End Date                                      
2021-06-16           21 CFR 1.502(a)      Develop FSVP   
2021-06-15           21 CFR 1.502(a)      Develop FSVP   
2021-06-09           21 CFR 1.502(a)      Develop FSVP   
2021-06-09           21 CFR 1.502(a)      Develop FSVP   
2021-06-07           21 CFR 1.502(a)      Develop FSVP   

                                 Long Description  \
Inspection End Date                                 
2021-06-16           You did not develop an FSVP.   
2021-06-15           You did not develop an FSVP.   
2021-06-09           You did not develop an FSVP.   
2021-06-09           You did not develop an FSVP.   
2021-06-07           You did not develop an FSVP.   

                                                          Firm Profile  
Inspection End Date                                                     
2021-06-16           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-15           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-09           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-09           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-07           https://datadashboard.fda.gov/ora/firmprofile....

In [132]:
Devices = insp[insp["Program Area"] == 'Devices']
Devices.head(5)

Legal Name Program Area  \
Inspection End Date                                                         
2021-06-10                                       ImaCor Inc.      Devices   
2021-06-10                                       ImaCor Inc.      Devices   
2021-06-10                                       ImaCor Inc.      Devices   
2021-06-04           Alliance Precision Plastics Corporation      Devices   
2021-06-04           Alliance Precision Plastics Corporation      Devices   

                        Act/CFR Number  \
Inspection End Date                      
2021-06-10               21 CFR 803.17   
2021-06-10            21 CFR 820.40(a)   
2021-06-10           21 CFR 820.198(a)   
2021-06-04            21 CFR 820.70(i)   
2021-06-04            21 CFR 820.80(c)   

                                                     Short Description  \
Inspection End Date                                                      
2021-06-10                              Lack of Written MDR Procedures   
2021-06-10                     Document review, approval documentation   
2021-06-10                                                  Complaints   
2021-06-04                 Software validation for automated processes   
2021-06-04           Lack of or inadequate In-process acceptance pr...   

                                                      Long Description  \
Inspection End Date                                                      
2021-06-10            Written MDR procedures have not been maintained.   
2021-06-10           The documentation of approval of documents doe...   
2021-06-10              Complaint files are not adequately maintained.   
2021-06-04           Software used as part of production has not be...   
2021-06-04           Procedures for the acceptance of in-process pr...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-06-10           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-10           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-10           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-04           https://datadashboard.fda.gov/ora/firmprofile....  
2021-06-04           https://datadashboard.fda.gov/ora/firmprofile....

In [133]:
Vet = insp[insp["Program Area"] == 'Veterinary Medicine']
Vet.head(5)

Legal Name         Program Area  \
Inspection End Date                                                  
2021-05-28                Bravo Packing, Inc.  Veterinary Medicine   
2021-05-28                Bravo Packing, Inc.  Veterinary Medicine   
2021-05-28                Bravo Packing, Inc.  Veterinary Medicine   
2021-05-28                Bravo Packing, Inc.  Veterinary Medicine   
2021-05-20           Lonsdale Feed Mill, Inc.  Veterinary Medicine   

                       Act/CFR Number  Short Description  \
Inspection End Date                                        
2021-05-28              21 CFR 507.14          Personnel   
2021-05-28           21 CFR 507.19(a)  Plant Maintenance   
2021-05-28           21 CFR 507.19(b)         Sanitation   
2021-05-28           21 CFR 507.19(e)       Pest Control   
2021-05-20           21 CFR 507.19(a)  Plant Maintenance   

                                                      Long Description  \
Inspection End Date                                                      
2021-05-28           You did not take reasonable measures and preca...   
2021-05-28           You did not maintain your plant in a clean man...   
2021-05-28           You did not clean, maintain and store utensils...   
2021-05-28           You did not take effective measures to exclude...   
2021-05-20           You did not maintain your plant in a clean man...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-05-28           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-28           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-28           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-28           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-20           https://datadashboard.fda.gov/ora/firmprofile....

In [134]:
Drugs = insp[insp["Program Area"] == 'Drugs']
Drugs.head(5)

Legal Name Program Area  \
Inspection End Date                                                       
2021-05-21           Fresenius Medical Care, North America        Drugs   
2021-05-06                          BRENNTAG PACIFIC, INC.        Drugs   
2021-05-06                          BRENNTAG PACIFIC, INC.        Drugs   
2021-05-06                          BRENNTAG PACIFIC, INC.        Drugs   
2021-05-04                        CMIC CMO USA Corporation        Drugs   

                        Act/CFR Number  \
Inspection End Date                      
2021-05-21              21 CFR 211.192   
2021-05-06            21 CFR 211.22(d)   
2021-05-06            21 CFR 211.25(a)   
2021-05-06           21 CFR 211.100(a)   
2021-05-04            21 CFR 211.67(c)   

                                              Short Description  \
Inspection End Date                                               
2021-05-21           Written record of investigation incomplete   
2021-05-06           Procedures not  in writing, fully followed   
2021-05-06            Training , Education , Experience overall   
2021-05-06                        Absence of Written Procedures   
2021-05-04                Cleaning/maintenance records not kept   

                                                      Long Description  \
Inspection End Date                                                      
2021-05-21           Written records of investigations into unexpla...   
2021-05-06           The responsibilities and procedures applicable...   
2021-05-06           Employees engaged in the manufacture, processi...   
2021-05-06           There are no written procedures for production...   
2021-05-04           Records are not kept for the maintenance and i...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-05-21           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-06           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-06           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-06           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-04           https://datadashboard.fda.gov/ora/firmprofile....

In [135]:
Bio = insp[insp["Program Area"] == 'Biologics']
Bio.head(5)

Legal Name  \
Inspection End Date                                                      
2021-05-18           U.S. Dept. of Veterans Affairs, Northport VA M...   
2021-05-18           U.S. Dept. of Veterans Affairs, Northport VA M...   
2021-02-24                                   New York Blood Center Inc   
2021-02-24                                   New York Blood Center Inc   
2021-02-24                                   New York Blood Center Inc   

                    Program Area     Act/CFR Number  \
Inspection End Date                                   
2021-05-18             Biologics   21 CFR 600.12(a)   
2021-05-18             Biologics  21 CFR 606.160(b)   
2021-02-24             Biologics   21 CFR 606.60(b)   
2021-02-24             Biologics  21 CFR 606.100(b)   
2021-02-24             Biologics  21 CFR 606.100(b)   

                                                     Short Description  \
Inspection End Date                                                      
2021-05-18                                  Maintenance  - concurrence   
2021-05-18                                            Required records   
2021-02-24                             Equipment calibration frequency   
2021-02-24           Establish, maintain and follow manufacturing SOPs   
2021-02-24                 Written SOPs available for use by personnel   

                                                      Long Description  \
Inspection End Date                                                      
2021-05-18           Records are not made concurrently with the per...   
2021-05-18           Failure to maintain processing and general rec...   
2021-02-24           Equipment used in the collection of blood and ...   
2021-02-24           Written standard operating procedures includin...   
2021-02-24           Failure to make available written procedures f...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-05-18           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-18           https://datadashboard.fda.gov/ora/firmprofile....  
2021-02-24           https://datadashboard.fda.gov/ora/firmprofile....  
2021-02-24           https://datadashboard.fda.gov/ora/firmprofile....  
2021-02-24           https://datadashboard.fda.gov/ora/firmprofile....

In [136]:
Bio_Research = insp[insp["Program Area"] == 'Bioresearch Monitoring']
Bio_Research.head(5)

Legal Name            Program Area  \
Inspection End Date                                               
2021-05-03           Chafic Karam, M.D.  Bioresearch Monitoring   
2021-05-03           Chafic Karam, M.D.  Bioresearch Monitoring   
2021-04-23            Robin Kroll, M.D.  Bioresearch Monitoring   
2021-04-23            Robin Kroll, M.D.  Bioresearch Monitoring   
2021-04-14             May T. Cho, M.D.  Bioresearch Monitoring   

                       Act/CFR Number  \
Inspection End Date                     
2021-05-03              21 CFR 312.60   
2021-05-03           21 CFR 312.62(b)   
2021-04-23              21 CFR 312.60   
2021-04-23           21 CFR 312.62(b)   
2021-04-14              21 CFR 312.60   

                                                  Short Description  \
Inspection End Date                                                   
2021-05-03                             FD-1572, protocol compliance   
2021-05-03           Case history records- inadequate or inadequate   
2021-04-23                             FD-1572, protocol compliance   
2021-04-23           Case history records- inadequate or inadequate   
2021-04-14                             FD-1572, protocol compliance   

                                                      Long Description  \
Inspection End Date                                                      
2021-05-03           An investigation was not conducted in accordan...   
2021-05-03           Failure to prepare or maintain accurate case h...   
2021-04-23           An investigation was not conducted in accordan...   
2021-04-23           Failure to prepare or maintain accurate case h...   
2021-04-14           An investigation was not conducted in accordan...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-05-03           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-03           https://datadashboard.fda.gov/ora/firmprofile....  
2021-04-23           https://datadashboard.fda.gov/ora/firmprofile....  
2021-04-23           https://datadashboard.fda.gov/ora/firmprofile....  
2021-04-14           https://datadashboard.fda.gov/ora/firmprofile....

In [137]:
Human_tiss = insp[insp["Program Area"] == 'Human Tissue for Transplantation']
Human_tiss.head(5)

Legal Name  \
Inspection End Date                                    
2021-03-16                Southwest Fertility Center   
2021-03-04           Caribbean Dental Products Corp.   
2020-10-16                                 Repro Lab   
2020-10-16                                 Repro Lab   
2020-10-16                                 Repro Lab   

                                         Program Area        Act/CFR Number  \
Inspection End Date                                                           
2021-03-16           Human Tissue for Transplantation     21 CFR 1271.85(a)   
2021-03-04           Human Tissue for Transplantation    21 CFR 1271.260(e)   
2020-10-16           Human Tissue for Transplantation     21 CFR 1271.47(a)   
2020-10-16           Human Tissue for Transplantation     21 CFR 1271.50(a)   
2020-10-16           Human Tissue for Transplantation  21 CFR 1271.55(b)(4)   

                                                    Short Description  \
Inspection End Date                                                     
2021-03-16                 Infection with communicable disease agents   
2021-03-04                  Storage temperatures recorded, maintained   
2020-10-16           Testing, screening, donor eligibility procedures   
2020-10-16               Determination based on screening and testing   
2020-10-16                       Reasons for ineligibility on summary   

                                                      Long Description  \
Inspection End Date                                                      
2021-03-16           Donors were not tested for evidence of infecti...   
2021-03-04           Storage temperatures of HCT/Ps were not record...   
2020-10-16           You did not establish and maintain procedures ...   
2020-10-16           HCT/P donors were not determined to be eligibl...   
2020-10-16           The summary of records for HCT/Ps from donors ...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-03-16           https://datadashboard.fda.gov/ora/firmprofile....  
2021-03-04           https://datadashboard.fda.gov/ora/firmprofile....  
2020-10-16           https://datadashboard.fda.gov/ora/firmprofile....  
2020-10-16           https://datadashboard.fda.gov/ora/firmprofile....  
2020-10-16           https://datadashboard.fda.gov/ora/firmprofile....

In [138]:
Part_11 = insp[insp["Program Area"] == 'Part 11 Compliance']
Part_11.head(5)

Legal Name  \
Inspection End Date                                                      
2021-02-25           Modesto Food Distributors dba Onorato & Compan...   
2021-01-21                    Wa Heng Dou Fu and Soy Sauce Corporation   
2021-01-20                                              Han Yang Corp.   
2021-01-15                                     D & M Smoked Fish, Inc.   
2020-08-20                                    Monica E. Kleinman, M.D.   

                           Program Area Act/CFR Number  \
Inspection End Date                                      
2021-02-25           Part 11 Compliance            NaN   
2021-01-21           Part 11 Compliance            NaN   
2021-01-20           Part 11 Compliance            NaN   
2021-01-15           Part 11 Compliance            NaN   
2020-08-20           Part 11 Compliance      21 CFR 11   

                           Short Description  \
Inspection End Date                            
2021-02-25           Procedural requirements   
2021-01-21           Procedural requirements   
2021-01-20           Procedural requirements   
2021-01-15           Procedural requirements   
2020-08-20                Electronic records   

                                                      Long Description  \
Inspection End Date                                                      
2021-02-25           Deviation from the procedural requirements of ...   
2021-01-21           Deviation from the procedural requirements of ...   
2021-01-20           Deviation from the procedural requirements of ...   
2021-01-15           Deviation from the procedural requirements of ...   
2020-08-20           Electronic records are used, but they do not m...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-02-25           https://datadashboard.fda.gov/ora/firmprofile....  
2021-01-21           https://datadashboard.fda.gov/ora/firmprofile....  
2021-01-20           https://datadashboard.fda.gov/ora/firmprofile....  
2021-01-15           https://datadashboard.fda.gov/ora/firmprofile....  
2020-08-20           https://datadashboard.fda.gov/ora/firmprofile....

In [139]:
Rad_Health = insp[insp["Program Area"] == 'Radiologic Health']
Rad_Health.head(5)

Legal Name  \
Inspection End Date                                           
2020-03-05                     Mevion Medical Systems, Inc.   
2020-03-05                     Mevion Medical Systems, Inc.   
2020-02-24           Technological Medical Advancements LLC   
2020-02-24           Technological Medical Advancements LLC   
2020-02-24           Technological Medical Advancements LLC   

                          Program Area            Act/CFR Number  \
Inspection End Date                                                
2020-03-05           Radiologic Health            21 CFR 1002.13   
2020-03-05           Radiologic Health      21 CFR 1002.30(a)(2)   
2020-02-24           Radiologic Health            21 CFR 1002.13   
2020-02-24           Radiologic Health          21 CFR 1010.2(b)   
2020-02-24           Radiologic Health  21 CFR 1040.10(f)(5)(ii)   

                                            Short Description  \
Inspection End Date                                             
2020-03-05                                  Failure to submit   
2020-03-05                                   Results of tests   
2020-02-24                                  Failure to submit   
2020-02-24           Certification other than by tag or label   
2020-02-24               Emission indicator - Class IIIb & IV   

                                                      Long Description  \
Inspection End Date                                                      
2020-03-05           You did not submit an annual report for produc...   
2020-03-05           You have not maintained records of the results...   
2020-02-24           You did not submit an annual report for produc...   
2020-02-24           Certification of a product is not in conforman...   
2020-02-24           An emission indicator which signals sufficient...   

                                                          Firm Profile  
Inspection End Date                                                     
2020-03-05           https://datadashboard.fda.gov/ora/firmprofile....  
2020-03-05           https://datadashboard.fda.gov/ora/firmprofile....  
2020-02-24           https://datadashboard.fda.gov/ora/firmprofile....  
2020-02-24           https://datadashboard.fda.gov/ora/firmprofile....  
2020-02-24           https://datadashboard.fda.gov/ora/firmprofile....

In [140]:
Parts_1240_1250 = insp[insp["Program Area"] == 'Parts 1240 and 1250']
Parts_1240_1250.head(5)

Legal Name  \
Inspection End Date                                             
2020-02-11                          Borenstein Caterers, Inc.   
2020-02-11                          Borenstein Caterers, Inc.   
2019-12-17                         New Orleans Aviation Board   
2019-12-12           Port Of Grays Harbor Terminal 1, 2 and 4   
2019-12-12                    Port of Grays Harbor Terminal 3   

                            Program Area     Act/CFR Number  \
Inspection End Date                                           
2020-02-11           Parts 1240 and 1250  21 CFR 1250.30(d)   
2020-02-11           Parts 1240 and 1250  21 CFR 1250.33(a)   
2019-12-17           Parts 1240 and 1250     21 CFR 1250.67   
2019-12-12           Parts 1240 and 1250     21 CFR 1250.67   
2019-12-12           Parts 1240 and 1250     21 CFR 1250.67   

                                              Short Description  \
Inspection End Date                                               
2020-02-11           Plumbing design, installation, maintenance   
2020-02-11                            Maintained in good repair   
2019-12-17                          Prevention of contamination   
2019-12-12                          Prevention of contamination   
2019-12-12                          Prevention of contamination   

                                                      Long Description  \
Inspection End Date                                                      
2020-02-11           Plumbing is not maintained so as to prevent co...   
2020-02-11           Not all working surfaces used in connection wi...   
2019-12-17           Failure to maintain servicing area piping syst...   
2019-12-12           Failure to construct and maintain servicing ar...   
2019-12-12           Failure to maintain servicing area hydrants in...   

                                                          Firm Profile  
Inspection End Date                                                     
2020-02-11           https://datadashboard.fda.gov/ora/firmprofile....  
2020-02-11           https://datadashboard.fda.gov/ora/firmprofile....  
2019-12-17           https://datadashboard.fda.gov/ora/firmprofile....  
2019-12-12           https://datadashboard.fda.gov/ora/firmprofile....  
2019-12-12           https://datadashboard.fda.gov/ora/firmprofile....

#### FOCUS ON BIOLOGICS DATA SUB-SET

In [141]:
print(Bio.shape, 'Rows, Columns')

(4153, 6) Rows, Columns


Above summary indicates 4153 entries for Biologics to date from 2008 to 2021

In [142]:
bio_unique_companies = Bio.drop_duplicates(subset = ["Legal Name"])
bio_unique_companies

Legal Name  \
Inspection End Date                                                      
2021-05-18           U.S. Dept. of Veterans Affairs, Northport VA M...   
2021-02-24                                   New York Blood Center Inc   
2020-11-09                           New York Blood Center Enterprises   
2020-10-21                                            BPC Plasma, Inc.   
2020-03-12                                                Hemarus, LLC   
...                                                                ...   
2008-12-08                              Depaul Community Health Center   
2008-12-05                Dorado Health, Inc.dba Manati Medical Center   
2008-12-04            Department of Air Force, 59th Medical Wing/SGVLB   
2008-11-04                             Memorial Hospital of South Bend   
2008-10-17           American Red Cross Blood Services Heart of Ame...   

                    Program Area     Act/CFR Number  \
Inspection End Date                                   
2021-05-18             Biologics   21 CFR 600.12(a)   
2021-02-24             Biologics   21 CFR 606.60(b)   
2020-11-09             Biologics  21 CFR 606.100(b)   
2020-10-21             Biologics   21 CFR 630.30(a)   
2020-03-12             Biologics   21 CFR 606.60(a)   
...                          ...                ...   
2008-12-08             Biologics  21 CFR 606.170(a)   
2008-12-05             Biologics  21 CFR 606.100(b)   
2008-12-04             Biologics   21 CFR 606.20(b)   
2008-11-04             Biologics   21 CFR 606.20(b)   
2008-10-17             Biologics  21 CFR 606.100(b)   

                                                     Short Description  \
Inspection End Date                                                      
2021-05-18                                  Maintenance  - concurrence   
2021-02-24                             Equipment calibration frequency   
2020-11-09           Establish, maintain and follow manufacturing SOPs   
2020-10-21                                        Donation suitability   
2020-03-12                Equipment observed, standardized, calibrated   
...                                                                ...   
2008-12-08                           Adverse Reaction - Investigations   
2008-12-05           Establish, maintain and follow manufacturing SOPs   
2008-12-04                     Qualifications of responsible personnel   
2008-11-04                     Qualifications of responsible personnel   
2008-10-17           Establish, maintain and follow manufacturing SOPs   

                                                      Long Description  \
Inspection End Date                                                      
2021-05-18           Records are not made concurrently with the per...   
2021-02-24           Equipment used in the collection of blood and ...   
2020-11-09           Written standard operating procedures includin...   
2020-10-21           Failure to determine the suitability of a dona...   
2020-03-12           Equipment used in the collection of blood and ...   
...                                                                ...   
2008-12-08           A thorough investigation of each reported adve...   
2008-12-05           Written standard operating procedures includin...   
2008-12-04           The personnel responsible for the processing a...   
2008-11-04           The personnel responsible for the compatibilit...   
2008-10-17           Written standard operating procedures includin...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-05-18           https://datadashboard.fda.gov/ora/firmprofile....  
2021-02-24           https://datadashboard.fda.gov/ora/firmprofile....  
2020-11-09           https://datadashboard.fda.gov/ora/firmprofile....  
2020-10-21           https://datadashboard.fda.gov/ora/firmprofile....  
2020-03-12       

In [143]:
#Convert Biologics Subset into a pandas dataframe
Bio=pd.DataFrame(Bio)
Bio

Legal Name  \
Inspection End Date                                                      
2021-05-18           U.S. Dept. of Veterans Affairs, Northport VA M...   
2021-05-18           U.S. Dept. of Veterans Affairs, Northport VA M...   
2021-02-24                                   New York Blood Center Inc   
2021-02-24                                   New York Blood Center Inc   
2021-02-24                                   New York Blood Center Inc   
...                                                                ...   
2008-10-17           American Red Cross Blood Services Heart of Ame...   
2008-10-17           American Red Cross Blood Services Heart of Ame...   
2008-10-10                                    Lonza Walkersville, Inc.   
2008-10-10                                    Lonza Walkersville, Inc.   
2008-10-06                           American Red Cross Blood Services   

                    Program Area       Act/CFR Number  \
Inspection End Date                                     
2021-05-18             Biologics     21 CFR 600.12(a)   
2021-05-18             Biologics    21 CFR 606.160(b)   
2021-02-24             Biologics     21 CFR 606.60(b)   
2021-02-24             Biologics    21 CFR 606.100(b)   
2021-02-24             Biologics    21 CFR 606.100(b)   
...                          ...                  ...   
2008-10-17             Biologics    21 CFR 606.100(c)   
2008-10-17             Biologics    21 CFR 606.170(a)   
2008-10-10             Biologics  21 CFR 600.14(a)(1)   
2008-10-10             Biologics     21 CFR 601.12(c)   
2008-10-06             Biologics    21 CFR 606.100(b)   

                                                     Short Description  \
Inspection End Date                                                      
2021-05-18                                  Maintenance  - concurrence   
2021-05-18                                            Required records   
2021-02-24                             Equipment calibration frequency   
2021-02-24           Establish, maintain and follow manufacturing SOPs   
2021-02-24                 Written SOPs available for use by personnel   
...                                                                ...   
2008-10-17                                     Thorough investigations   
2008-10-17                           Adverse Reaction - Investigations   
2008-10-10                              Who must report - manufacturer   
2008-10-10                 Moderate changes to an approved application   
2008-10-06           Establish, maintain and follow manufacturing SOPs   

                                                      Long Description  \
Inspection End Date                                                      
2021-05-18           Records are not made concurrently with the per...   
2021-05-18           Failure to maintain processing and general rec...   
2021-02-24           Equipment used in the collection of blood and ...   
2021-02-24           Written standard operating procedures includin...   
2021-02-24           Failure to make available written procedures f...   
...                                                                ...   
2008-10-17           Failure to perform a thorough investigation of...   
2008-10-17           A thorough investigation of each reported adve...   
2008-10-10             Failure to submit biological deviation reports.   
2008-10-10           Failure to submit a supplement prior to distri...   
2008-10-06           Written standard operating procedures includin...   

                                                          Firm Profile  
Inspection End Date                                                     
2021-05-18           https://datadashboard.fda.gov/ora/firmprofile....  
2021-05-18           https://datadashboard.fda.gov/ora/firmprofile....  
2021-02-24           https://datadashboard.fda.gov/ora/firmprofile....  
2021-02-24           https://datadashboard.fda.gov/ora/firmprofi

In [150]:
Bio['2020']

<ipython-input-150-43763714b8f3>:1: FutureWarning: Indexing a DataFrame with a datetimelike index using a single string to slice the rows, like `frame[string]`, is deprecated and will be removed in a future version. Use `frame.loc[string]` instead.
  Bio['2020']


Legal Name  \
Inspection End Date                                                      
2020-11-09                           New York Blood Center Enterprises   
2020-10-21                                            BPC Plasma, Inc.   
2020-03-12                                                Hemarus, LLC   
2020-03-12                                                Hemarus, LLC   
2020-03-12                                                Hemarus, LLC   
2020-03-04                            The Community Blood Center, Inc.   
2020-02-27                                       Millipore (U.K.) Ltd.   
2020-02-27                                       Millipore (U.K.) Ltd.   
2020-02-27                               American Red Cross Blood Svcs   
2020-02-27                               American Red Cross Blood Svcs   
2020-02-26                                 NCH Baker Hospital Downtown   
2020-02-25                                                 Genesiscare   
2020-02-25                                                 Genesiscare   
2020-02-13           Biological Specialty Company, LLC dba BioIVT, ...   
2020-02-13           Biological Specialty Company, LLC dba BioIVT, ...   
2020-02-12                                             CSL Plasma, Inc   
2020-02-12                                             CSL Plasma, Inc   
2020-02-11           Blood Bank Of Delmarva, a Division of New York...   
2020-02-11           Blood Bank Of Delmarva, a Division of New York...   
2020-02-11                                             Biomat USA, Inc   
2020-02-11                                             Biomat USA, Inc   
2020-01-24                                            Aultman Hospital   
2020-01-17                                        Blood Assurance Inc.   
2020-01-15                                    Lexington Medical Center   

                    Program Area        Act/CFR Number  \
Inspection End Date                                      
2020-11-09             Biologics     21 CFR 606.100(b)   
2020-10-21             Biologics      21 CFR 630.30(a)   
2020-03-12             Biologics      21 CFR 606.60(a)   
2020-03-12             Biologics   21 CFR 630.40(b)(3)   
2020-03-12             Biologics      21 CFR 640.64(e)   
2020-03-04             Biologics     21 CFR 606.100(c)   
2020-02-27             Biologics      21 CFR 600.12(a)   
2020-02-27             Biologics          21 CFR 610.1   
2020-02-27             Biologics         21 CFR 606.40   
2020-02-27             Biologics      21 CFR 606.60(a)   
2020-02-26             Biologics  21 CFR 606.160(a)(1)   
2020-02-25             Biologics     21 CFR 606.100(b)   
2020-02-25             Biologics  21 CFR 606.160(a)(1)   
2020-02-13             Biologics         21 CFR 606.65   
2020-02-13             Biologics  21 CFR 606.160(a)(1)   
2020-02-12             Biologics        21 CFR 606.171   
2020-02-12             Biologics      21 CFR 630.10(c)   
2020-02-11             Biologics   21 CFR 630.40(b)(1)   
2020-02-11             Biologics   21 CFR 630.40(b)(4)   
2020-02-11             Biologics   21 CFR 606.40(a)(1)   
2020-02-11             Biologics      21 CFR 640.64(e)   
2020-01-24             Biologics     21 CFR 606.100(b)   
2020-01-17             Biologics   21 CFR 606.40(a)(1)   
2020-01-15             Biologics      21 CFR 606.60(a)   

                                                     Short Description  \
Inspection End Date                                                      
2020-11-09           Establish, maintain and follow manufacturing SOPs   
2020-10-21                                        Donation suitability   
2020-03-12                Equipment observed, standardized, calibrated   
2020-03-12                                          Results of testing   
2020-03-12                                 Prevention of contamination   
2020-03-04                                     Thorough investigations   
2020-02-27                        